### PR TITLE
Disabling incremental/ich_method_call_trait_scope.rs

### PR DESCRIPTION
**Pull request by sophiajt**
This disables the `incremental/ich_method_call_trait_scope.rs` in the hopes we can get the builds up and running again.

r? @nikomatsakis

### CODE REVIEW

**Comment by jonathandturner** [comment]
@bors r=nikomatsakis p=10

Approving at Niko's request

**Comment by nikomatsakis** [comment]
😿

**Comment by alexcrichton** [comment]
Can we not ignore tests like this? This is basically a death sentence for this test history has shown.

This kind of error comes up when file names are too long, and the test name here is pretty long. Looking more into the paths, this path is on the bots:

```
/c/bot/slave/auto-win-msvc-64-opt-rustbuild/build/obj/build/x86_64-pc-windows-msvc/test/incremental/ich_method_call_trait_scope.incremental/ich_method_call_trait_scope-224534f2bd4fd5de/sess-53af0b4562f62-20bf3884-9ace1227172865df/cgu-ich_method_call_trait_scope-mod3.o
```

which is 268 bytes long, suspiciously close to the 260 limit in Windows.

Can we just rename the test to a shorter name?

**Comment by jonathandturner** [comment]
@alexcrichton - 82 characters of that is the path leading into the source tree.  Does it need to be that long?

**Comment by jonathandturner** [comment]
I guess some of that comes from the build system putting the system triple in the path...

**Comment by jonathandturner** [comment]
Also... "224534f2bd4fd5de/sess-53af0b4562f62-20bf3884-9ace1227172865df"

**Comment by nikomatsakis** [comment]
OK, so, that path includes @michaelwoerister's "session directories", which are not yet landed -- that is, that path is not being used *yet*. (But the PR that adds them @michaelwoerister is trying to land right now)

**Comment by alexcrichton** [comment]
None of that's as easy to change as the test name, presumably all other incremental tests have the same relative length of path and are working.

Yes leading up to the build directory is long and we should fix that, but in the meantime we can just write shorter test names. Yes the incremental directory takes up a lot of space, and we should fix that too.

**Comment by jonathandturner** [comment]
@nikomatsakis - does that mean the session directories might cause other issues on windows?

**Comment by nikomatsakis** [comment]
@jonathandturner perhaps, yes. I certainly agree with @alexcrichton that skipping tests is a recipe for them being overlooked. I was thinking that a nice thing to do would be to add more instrumentation to compiletest for unwraps so it at least prints the path of the file in question (I feel like this is nigh impossible to debug otherwise...)

At minimum, we ought to open an issue I guess and set it to P-high. (I can do that.)

**Comment by nikomatsakis** [comment]
https://github.com/rust-lang/rust/issues/36020

**Comment by alexcrichton** [comment]
Ok, so after thinking a bit more I don't think we need to land this. The problem here is likely some errant `fs::remove_dir_all` somewhere in compiletest trying to clean up directories. The directory has a super long path name in it which then fails to get removed. The file was added by @michaelwoerister's PR adding session directories, but it bounced, so we couldn't clearly trace it back to that.

tl;dr; this is a problem with dirty build directories. I've manually cleaned out the ones on the bots so we shouldn't need to ignore the test.

@bors: r- force

**Comment by michaelwoerister** [comment]
ALso note, I'll work on making the directory names shorter before trying to land the offending PR again:
https://github.com/rust-lang/rust/pull/35718#issuecomment-242815067
